libpromises/files_names.c

```diff
@@ -169,7 +166,7 @@ char *JoinSuffix(char *path, const char *leaf)
         return NULL;
     }
 
-    strcat(path, leaf);
+    strlcat(path, leaf, path_size);
```

Comment:
The length check was broken (assuming paht is CF_BUFSIZE).
However, as we have path_size now I reverted to using that and your favourite function strcat(). :)